### ISSUE TITLE
Templated Function calls Inconsistent highlighting

### ISSUE THREAD

**Issue by rbx** [jeff-hykin/cpp-textmate-grammar#87]
Not sure if this is a duplicate of e.g. #39, sorry if it is.

![Screenshot_2019-04-11_10-49-26](https://user-images.githubusercontent.com/1915061/55943796-8a0c3500-5c47-11e9-9f6e-a3c6d57370b4.png)

Is there a way to have `s` be the same color across all occurrences? What about the `a`?

Thanks!

**Comment by matter123** [comment]
Hi, this is a duplicate of #81. Would you mind posting your image on that one?

**Comment by rbx** [comment]
No problem, will do.
Meanwhile I have also discovered this:
![Screenshot_2019-04-11_16-11-06](https://user-images.githubusercontent.com/1915061/55964160-8347e700-5c74-11e9-9b0b-7c0bea8104fd.png)

in the first two lines `string` is the `std::string` type (so, not a variable as in #81?).
As you can see in the last two lines it works correctly if i prefix the types.
But then also the `putIfExists` function is highlighted differently...

**Comment by matter123** [comment]
This is blocked by #80 but I'll try to get a workaround today.

**Comment by matter123** [comment]
Okay, I have a solution and am creating a PR for it.

![Screenshot from 2019-04-11 12-03-09](https://user-images.githubusercontent.com/714007/55984334-16235a00-5c52-11e9-9266-49edff3ecc9d.png)

If you want to try this out now, you can replace `<vscode ext dir>/jeff-hykin.better-cpp-syntax-1.7.7/syntaxes/cpp.tmLanguage.json` with the file at https://raw.githubusercontent.com/jeff-hykin/cpp-textmate-grammar/Fix-%2387/syntaxes/cpp.tmLanguage.json
